mypy_django_plugin/transformers/managers.py

```diff
@@ -121,7 +122,7 @@ def _process_dynamic_method(
     is_fallback_queryset: bool,
 ) -> CallableType:
     variables = method_type.variables
-    ret_type = method_type.ret_type
+    ret_type = copy_type(method_type.ret_type)
```

Comment:
I think calling `get_proper_type` here will resolve the first mypy complaint from CI

```suggestion
    ret_type = copy_type(get_proper_type(method_type.ret_type))
```

Reply:
👍 
I also realised that making the copy inside _process_dynamic_method won't work with recursion (types like `list[list[T]]`), since copy_type only does a shallow copy.